libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -350,12 +350,18 @@ bool ASTJsonConverter::visit(OverrideSpecifier const& _node)
 
 bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 {
+	Visibility visibility;
+	if (_node.isConstructor())
+		visibility = _node.annotation().contract->abstract() ? Visibility::Internal : Visibility::Public;
+	else
+		visibility = _node.visibility();
```

Comment:
We could consider removing the visibility field for constructors, mainly because it is redundant.

Reply:
My suggestion is to set it to `null` so the field won't be exported. For the `ASTJsonImporter`, it should already work.

Reply:
I added this because it maintains backwards-compatibility.